extensions/wikia/Flags/maintenance/MoveNotices.php

```diff
@@ -0,0 +1,510 @@
+<?php
+
+$dir = dirname( __FILE__ ) . "/../../../../";
+$extDir = __DIR__ . "/../";
+
+require_once( $dir . 'includes/wikia/nirvana/WikiaObject.class.php' );
+require_once( $dir . 'includes/wikia/nirvana/WikiaModel.class.php' );
+require_once( $extDir . 'models/FlagsBaseModel.class.php' );
+require_once( $extDir . 'models/Flag.class.php' );
+require_once( $extDir . 'models/FlagType.class.php' );
+require_once( $dir . 'maintenance/Maintenance.php' );
+
+use Flags\Models\Flag;
+use Flags\Models\FlagType;
+
+class MoveNotices extends Maintenance {
+
+	/**
+	 * Set script options
+	 */
+	public function __construct() {
+		parent::__construct();
+		$this->addOption( 'csv', 'CSV file with data' );
+		$this->addOption( 'list', 'Run script without adding data to database' );
+		$this->addOption( 'section', 'Search template in given section. Otherwise all article is searched.' );
+	}
+
+	public function execute() {
+		global $wgCityId, $wgParser;
+
+		$csv = $this->getOption( 'csv' );
+		$section = $this->getOption( 'section' );
+		$list = $this->getOption( 'list' );
+
+		if ( empty( $csv ) ) {
+			$this->output( "You must attach CSV file.\n" );
+			return;
+		}
+
+		if ( !file_exists( $csv ) ) {
+			$this->output( "File $csv does not exist.\n" );
+			return;
+		}
+
+		// Prepare log file
+		$logName = substr( $csv, 0, strrpos( $csv, '.' ) );
+		$logName .= '.log';
+
+		$csvFile = fopen( $csv, 'r' );
+		$logFile = fopen( $logName, 'w+' );
+
+		$csvData = [];
+
+		if ( !$csvFile ) {
+			$this->output( "Cannot read file: $csv" );
+			return;
+		}
+
+		// Get data from scv file
+		while( ( $data = fgetcsv( $csvFile ) ) !== false ) {
+			$csvData[] = $data;
+		}
+
+		fclose( $csvFile );
+
+		$this->output( "Start processing\n" );
+
+		$flag = new Flag();
+		$flagTypes = new FlagType();
```

Comment:
Could you name them `$flagType**Model**` or similar? You're using `$flagTypes` array a few lines below and it may be confusing.